tests/modules/programs/zsh/history-path-new-custom.nix

```diff
@@ -0,0 +1,17 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+{
+  config = {
+    home.stateVersion = "20.03";
+    programs.zsh = {
+      enable = true;
+      history.path = "$HOME/some/directory/zsh_history";
+    };
+
+    nmt.script = ''
+      assertFileRegex home-files/.zshrc '^HISTFILE="$HOME/some/directory/zsh_history"$'
```

Comment:
No need to escape the `$` in `$HOME`?

Reply:
The test passes, so I guess no? When I add `echo '^HISTFILE="$HOME/some/directory/zsh_history"$'` it also prints with $HOME as it should. I guess it's because of the single quotes?

Reply:
Seems grep interprets $ literally if it's not at the end. Didn't know that 🙂

```console
$ grep '^foo$bar$' <<<'foobar'
$ grep '^foo$bar$' <<<'foo$bar'
foo$bar 
```